### DISCUSSION TITLE
Is it possible to make the Undo command behave like normal Vim?

### DISCUSSION THREAD

**Discussion by nathanielwarner** [JetBrains/ideavim#310] [Q&A]
I like this plugin a lot, but there's a rough edge that's really annoying me. In Vim, if I enter Insert mode, insert some text, go back to normal mode, and then undo with "u", all the text I inserted is removed. But with IdeaVim, the "u" command acts like the IntelliJ undo "Ctrl-Z", meaning that it only undoes a word or two at a time. It seems like the insertion of text is not considered one "action", so undoing it does not undo it all at once.

Does anyone know how to work around this? I just want it to behave like regular Vim, where Undo actually removes all the text I just inserted.

**Comment by AlexPl292** [comment]
Hi! Unfortunately, it's not yet implemented. Here is a pretty old issue regarding that, but over the time it became one of the most voted: https://youtrack.jetbrains.com/issue/VIM-308

**Comment by rmcfarlane82** [comment]
Stumbled across this, not sure if this helps but in the end I ended up adding this to my ideavimrc 
```
inoremap <Space> <C-g>u<Space>
inoremap . <C-g>u.
inoremap , <C-g>u,
inoremap <CR> <C-g>u<CR>
```
basically every time it sees a space, period, comma or a carriage return it tells vim/ideavim to add a new undo step.
this minimizes the amount it deletes when pressing u
